ui/react-components/components/AddAppointment/AddAppointment.jsx

```diff
@@ -256,51 +249,12 @@ const AppointmentEditor = props => {
-                    <div data-testid="patient-search">
-                        <PatientSearch
-                            onChange={(optionSelected) => {
-                            const newValue = optionSelected ? optionSelected.value : undefined;
-                            updateAppointmentDetails({patient: newValue});
-                            updateErrorIndicators({patientError: !newValue});
-                        }}/>
-                        <ErrorMessage message={errors.patientError ? errorTranslations.patientErrorMessage : undefined}/>
-                    </div>
-                    <div data-testid="service-search">
-                        <ServiceSearch onChange={(optionSelected) => {
-                            updateAppointmentDetails({service: optionSelected.value});
-                            updateErrorIndicators({serviceError: !optionSelected.value});
-                            endTimeBasedOnService(appointmentDetails.startTime, optionSelected.value, undefined);
-                        }}
-                                       specialityUuid={appointmentDetails.speciality}/>
-                        <ErrorMessage message={errors.serviceError ? errorTranslations.serviceErrorMessage : undefined}/>
-                    </div>
-                    <div data-testid="service-type-search">
-                        <ServiceTypeSearch onChange={(optionSelected) => {
-                            updateAppointmentDetails({serviceType: optionSelected.value});
-                            endTimeBasedOnService(appointmentDetails.startTime, undefined, optionSelected.value);
-                        }}
-                       serviceUuid={appointmentDetails.service && appointmentDetails.service.uuid}/>
-                    </div>
-                    {isSpecialitiesEnabled(appConfig) ?
-                        <div data-testid="speciality-search">
-                            <SpecialitySearch onChange={(optionSelected) => updateAppointmentDetails({speciality: optionSelected.value})}/>
-                        </div> : null
-                    }
-                    <div data-testid="location-search">
-                        <LocationSearch onChange={(optionSelected) => updateAppointmentDetails({location: optionSelected.value})}/>
-                        <ErrorMessage message={undefined}/>
-                    </div>
-                </div>
-                <div className={classNames(searchFieldsContainerRight)} data-testid="provider-search">
-                    <ProviderSearch onChange={selectedProviders => updateAppointmentDetails({providers: selectedProviders})}
-                                    maxAppointmentProvidersAllowed={maxAppointmentProvidersAllowed(appConfig)}/>
-                </div>
-            </div>
+            <SearchFieldsContainer updateAppointmentDetails={updateAppointmentDetails} updateErrorIndicators={updateErrorIndicators}
```

Comment:
This makes it more difficult to understand the code?? Do we really need this?

Reply:
This snippet is common Add and Edit appointment. So we extracted that part.